File: Eclipse Projects/webapps/TeleDirectPhp/WebContent/scripts/php/Test/TestSecurity.php
<?php
require_once 'Security.php';

class TestSecurity extends PHPUnit_Framework_TestCase{
	protected $security, $username, $password, $token;
	
	protected function setUp(){
		$this->security = new Security();
		$this->username = "akash";
		$this->password = "akash";
	}
	
	public function testLogin(){
		
		$actualResponse = json_decode($this->security->login($this->username, $this->password));
		
		$expectedResponse = json_decode(json_encode(array(
			"status" => "success",
			"message" => "Welcome, Akash"
		)));
		
		$this->assertEquals($expectedResponse->status, $actualResponse->status);
		$this->assertEquals($expectedResponse->message, $actualResponse->message);
		
		$this->token = $actualResponse->data->session->token;
	}
	
	
	public function testValid(){
		$actualResponse = json_decode($this->security->valid($this->username, $this->token));
		
		$expectedResponse = json_decode(json_encode(array(
				"status" => "failure",
				"message" => "Session not valid"
		)));
		
		$this->assertEquals($expectedResponse->status, $actualResponse->status);
		$this->assertEquals($expectedResponse->message, $actualResponse->message);
	}
	
	public function testLogout(){
		$this->username = "akash";
	
		$actualResponse = json_decode($this->security->logout($this->username));
	
		$expectedResponse = json_decode(json_encode(array(
				"status" => "success",
				"message" => "akash has logged out successfully."
		)));
	
		$this->assertEquals($expectedResponse->status, $actualResponse->status);
		$this->assertEquals($expectedResponse->message, $actualResponse->message);
	}
	
	
}
?>

File: Eclipse Projects/webapps/TeleDirectJavaServer/src/com/test/teleDirect/TestSecurity.java
package com.test.teleDirect;

import static org.junit.Assert.*;

import java.util.UUID;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import com.teleDirect.Security;
import com.teleDirect.Session;
import com.teleDirect.server.Response;
import com.teleDirect.server.ServerCoordinator;

public class TestSecurity {

	Security security;
	
	@Before
    public void setUp() {
        security = new Security();
    }

    @After
    public void tearDown() {
    	//after execution
    }
    
	@Test
	public void shouldBeAbleToValidateSession() {
		String username = "akash";
		String password = "akash";
		String token = ServerCoordinator.convertParamsToString(UUID.randomUUID().toString());
		Response response;
		
		response = security.validateSession(username, token);
		assertEquals("failure", response.getStatus());
		assertEquals("Session not valid.", response.getMessage());
		
		response = security.login(username, password);
		Session session = (Session) response.getData();
		token = session.getToken();
		response = security.validateSession(username, token);
		assertEquals("success", response.getStatus());
		assertEquals("Sesssion is valid.", response.getMessage());
		
	}
	
	@Test
	public void shouldBeAbleToLogin() {
		Response response;
		String username = "akash";
		String password = "akash";
		
		response = security.login(username, password);
		assertEquals("success", response.getStatus());
		assertEquals("Welcome, " + username, response.getMessage());
		assertEquals(username, ((Session)response.getData()).getUser().getUsername());
		
		username = "akash";
		password = "akassh";
		response = security.login(username, password);
		assertEquals("failure", response.getStatus());
		assertEquals("Wrong username/password", response.getMessage());
		assertNull(response.getData());
	}
	
	@Test
	public void shouldBeAbleToLogout() {
		Response response;
		String username = "akash";
		
		response = security.logout(username);
		assertEquals("success", response.getStatus());
		assertEquals(username + " has logged out successfully.", response.getMessage());
	}

}
